core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2433,11 +2433,12 @@ private[spark] object Utils extends Logging {
    * if the key is not set in the Hadoop configuration.
    */
   def getSparkOrYarnConfig(conf: SparkConf, key: String, default: String): String = {
-    val sparkValue = conf.get(key, default)
-    if (conf.get(SparkLauncher.SPARK_MASTER, null) == "yarn") {
-      new YarnConfiguration(SparkHadoopUtil.get.newConfiguration(conf)).get(key, sparkValue)
+    if (conf.contains(key)) {
```

Comment:
The scaladoc above is now wrong, since it still refers to the old order of precedence. Otherwise looks ok.